app/controllers/api/grade_entry_forms_controller.rb

```diff
@@ -6,7 +6,7 @@ class GradeEntryFormsController < MainApiController
     # Requires: id
     def show
       grade_entry_form = record
-      send_data grade_entry_form.export_as_csv,
+      send_data grade_entry_form.export_as_csv(@current_role),
```

Comment:
use `current_role` in controllers. You only need to access the instance variable directly in views because views don't have access to session methods.